cpp/src/arrow/util/macros.h

```diff
@@ -22,7 +22,7 @@
 #ifndef ARROW_DISALLOW_COPY_AND_ASSIGN
 #define ARROW_DISALLOW_COPY_AND_ASSIGN(TypeName) \
   TypeName(const TypeName&) = delete;            \
-  TypeName& operator=(const TypeName&) = delete
+  void operator=(const TypeName&) = delete
```

Comment:
Hi, @wesm why change `TypeName` to `void`?

Reply:
I don't remember why. Changing it back to a reference type would probably be a good idea, can you open a Jira issue?

https://en.wikipedia.org/wiki/Assignment_operator_(C%2B%2B)#Return_value_of_overloaded_assignment_operator

Reply:
> 

My pleasure.